### PR TITLE
policy: use netip.Addr when constructing CIDR rules

### PR DESCRIPTION
**Pull request by tklauser**
First commit adds a helper function, second commit converts CIDR rules to use netip.Addr internally. This should somewhat improve memory usage when using CIDR rules in policies.

Before/after benchmark:
    
    name          old time/op    new time/op    delta
    GetCIDRSet-8    2.58µs ± 6%    1.96µs ± 6%  -23.84%  (p=0.000 n=20+20)
    
    name          old alloc/op   new alloc/op   delta
    GetCIDRSet-8      608B ± 0%      416B ± 0%  -31.58%  (p=0.000 n=20+20)
    
    name          old allocs/op  new allocs/op  delta
    GetCIDRSet-8      13.0 ± 0%      11.0 ± 0%  -15.38%  (p=0.000 n=20+20)

See commits for details.

### CODE REVIEW

File: pkg/ip/ip.go [hunk 1]
```diff
@@ -764,6 +764,19 @@ func KeepUniqueIPs(ips []net.IP) []net.IP {
 	return returnIPs
 }
 
+func KeepUniqueAddrs(addrs []netip.Addr) []netip.Addr {
```
Comment (ldelossa) [inline pkg/ip/ip.go:773]: Do we want to perform a copy first then sort?

If not, we should comment this so the caller knows their original `addrs` array will be modified (sorted, in this case).

Alternatively, you can make this function take `addrs *[]netip.Addr` and return no value, which indicates to the caller the array is modified completely in place. The returned slice is always < len(addrs) so I believe you won't strictly need to return `[]netip.Addr` 

If the function receives a pointer to a slice you can then update the pointer's value to the new slice 

For example line 769 becomes
```
*addrs = (*addrs)[:0]
```

Comment (tklauser) [reply pkg/ip/ip.go:773]: I've tried to keep this as close as possible to `KeepUniqueIPs` which modifies the original `addrs` slice. The way that function is currently used, the in-place modification and sorting is expected. Thus, I'd keep it as is for now and add a godoc comment describing its behavior.

Comment (ldelossa) [reply pkg/ip/ip.go:773]: lgtm

**Review by ldelossa — APPROVED** [review]

**Comment by tklauser** [comment]
/test-1.25-net-next

**Review by nebril — APPROVED** [review]

**Review by rolinh — APPROVED** [review]
lgtm (no before/after benchmarks? :slightly_smiling_face: )

**Comment by tklauser** [comment]
> lgtm (no before/after benchmarks? slightly_smiling_face )

Good point. Added benchmarks to the last commit and the PR description.

**Comment by tklauser** [comment]
/test-1.16-4.9

**Comment by tklauser** [comment]
/test-1.25-net-next

**Comment by tklauser** [comment]
/test-1.25-net-next

**Review by gandro — APPROVED** [review]

**Pull request merged**